src/pages/AppDetails/AppDetailsPage.tsx

```diff
@@ -48,22 +48,19 @@ class AppDetails extends React.Component<RouteComponentProps<AppId>, AppDetailsS
   }
 
   fetchApp = () => {
-    const promiseDetails = API.getApp(authentication(), this.props.match.params.namespace, this.props.match.params.app);
-
-    const promiseHealth = API.getAppHealth(
-      authentication(),
-      this.props.match.params.namespace,
-      this.props.match.params.app,
-      600
-    );
-
-    Promise.all([promiseDetails, promiseHealth])
-      .then(([resultDetails, resultHealth]) => {
-        this.setState({
-          app: resultDetails.data,
-          health: resultHealth
-        });
+    API.getApp(authentication(), this.props.match.params.namespace, this.props.match.params.app)
```

Comment:
I see.
It will improve the scenario where there are no sidecars, but when there is, queries are sequentally instead to be in parallel.
Probably not a big deal, but yeah, in that case we will add more latency.

Reply:
But it's just an observation, this pattern is needed, as it needs the "hasSidecar" as input.

Reply:
@lucasponce that's correct, but on the other hand it's now going to set the state right after the details are fetched (without waiting for health) so it's also an improvement from this perspective. But yes, it's not done in parallel anymore.